src/App.tsx

```diff
@@ -15,71 +15,85 @@ export const goods = [
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  const handleGoodSelect = (good: string) => {
+    setState({ selectedGood: good });
+  };
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+  const handleClearSelection = () => {
+    setState({ selectedGood: '' });
+  };
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+  const isGoodSelected = (good: string) => {
+    return state.selectedGood === good;
+  };
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+  return (
+    <main className="section container">
+      <h1 className="title">
+        {state.selectedGood ? `${state.selectedGood} is selected` : 'No goods selected'}
+      </h1>
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+      {state.selectedGood && (
+        <h1 className="title is-flex is-align-items-center">
+          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+          <button
+            data-cy="ClearButton"
+            type="button"
+            className="delete ml-3"
+            onClick={handleClearSelection}
+          />
+        </h1>
+      )}
 
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
+      <table className="table">
+        <tbody>
+          {goods.map((good) => (
+            <tr
+              data-cy="Good"
+              key={good}
+              className={isGoodSelected(good)
+                ? 'has-background-success-light' : ''}
             >
-              +
-            </button>
-          </td>
+              <td>
+                {!isGoodSelected(good) && (
+                  <button
+                    data-cy="AddButton"
+                    type="button"
+                    className="button"
+                    onClick={() => handleGoodSelect(good)}
+                  >
+                    +
+                  </button>
+                )}
+
+                {isGoodSelected(good) && (
+                  <button
+                    data-cy="RemoveButton"
+                    type="button"
+                    className="button is-info"
+                    onClick={handleClearSelection}
+                  >
+                    -
+                  </button>
+                )}
```

Comment:
also you can make one button and conditionally render data-cy, className, onClick, and button textContent